tests/integration/test_charm.py

```diff
@@ -46,9 +46,10 @@ async def test_build_and_deploy(ops_test):
     await ops_test.model.wait_for_idle(
         apps=[MLMD], status="active", raise_on_blocked=False, idle_period=30
     )
-    await ops_test.model.wait_for_idle(
-        apps=[APP_NAME], status="blocked", raise_on_blocked=False, idle_period=30
-    )
+    # TODO: Restore this if we decide Ingress is a required relation for the charm
+    # await ops_test.model.wait_for_idle(
+    #     apps=[APP_NAME], status="blocked", raise_on_blocked=False, idle_period=30
+    # )
```

Comment:
I think we can remove it entirely, a unit test would be sufficient for testing the unit status in the absence of a required relation.

Reply:
Good point.  Removed.

Also, realized when I was removing this that the `build_and_deploy` never checks that envoy has actually deployed successfully.  Added that in here too